app/tests/test_download.py

```diff
@@ -158,6 +159,66 @@ def test_download_record_standard_user_without_citable_reference(
         assert caplog.records[0].levelname == "INFO"
         assert caplog.records[0].message == msg
 
+    @mock_aws
+    def test_download_record_standard_user_get_file_errors(
+        self, app, client, mock_standard_user
+    ):
+        """
+        Given a file is requested from the database / S3 which doesn't exist
+        When a standard user tries to access the file to download
+        Then the response status code should be 404
+        """
+
+        bucket_name = "test_bucket"
+        file = FileFactory(
+            FileType="file", FileName="testimage.png", CiteableReference=None
+        )
+        create_mock_s3_bucket_with_object(bucket_name, file)
+        app.config["RECORD_BUCKET_NAME"] = bucket_name
+
+        mock_standard_user(client, file.consignment.series.body.Name)
+        response = client.get(f"{self.route_url}/invalid_file")
+
+        assert response.status_code == 404
+
+    @mock_aws
+    @patch("boto3.client")
+    def test_download_record_standard_user_read_file_error(
+        self, mock_boto3_client, app, client, mock_standard_user, caplog
+    ):
+        """
+        Given a file in the database with corresponding file in the S3 bucket
+            but reading the file content fails
+        When a standard user with access to the file's transferring body makes a request to download the record
+        Then the response status code should be 500
+        """
+
+        bucket_name = "test_bucket"
+        file = FileFactory(
+            FileType="file", FileName="testimage.png", CiteableReference=None
+        )
+        create_mock_s3_bucket_with_object(bucket_name, file)
+        app.config["RECORD_BUCKET_NAME"] = bucket_name
+
+        # Mock the S3 client and its get_object method
+        mock_s3_client = MagicMock()
```

Comment:
If youre using moto I dont think you need to create magic mocks for s3. You could use the moto api to put an object: https://docs.getmoto.org/en/latest/docs/getting_started.html#moto-usage

But happy to leave as is if this works